python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -827,26 +828,48 @@ class DbtCliResource(ConfigurableResource):
         ),
     )
     profile: Optional[str] = Field(
-        default=None,
+        # default=None,
         description=(
             "The profile from your dbt `profiles.yml` to use for execution. See"
             " https://docs.getdbt.com/docs/core/connect-data-platform/connection-profiles for more"
             " information."
         ),
     )
     target: Optional[str] = Field(
-        default=None,
+        # default=None,
         description=(
             "The target from your dbt `profiles.yml` to use for execution. See"
             " https://docs.getdbt.com/docs/core/connect-data-platform/connection-profiles for more"
             " information."
         ),
     )
     dbt_executable: str = Field(
-        default=DBT_EXECUTABLE,
+        # default=DBT_EXECUTABLE,
         description="The path to the dbt executable.",
     )
 
+    def __init__(
+        self,
+        project_dir: Union[str, DbtProject],
+        global_config_flags: Optional[List[str]] = None,
+        profiles_dir: Optional[str] = None,
+        profile: Optional[str] = None,
+        target: Optional[str] = None,
+        dbt_executable: str = DBT_EXECUTABLE,
+    ):
+        if isinstance(project_dir, DbtProject):
+            project_dir = os.fspath(project_dir.project_dir)
+
+        # static typing doesn't understand whats going on here, thinks these fields dont exist
+        super().__init__(
+            project_dir=project_dir,  # type: ignore
+            global_config_flags=global_config_flags or [],  # type: ignore
+            profiles_dir=profiles_dir,  # type: ignore
+            profile=profile,  # type: ignore
+            target=target,  # type: ignore
+            dbt_executable=dbt_executable,  # type: ignore
+        )
```

Comment:
[1]

Reply:
`__init__` is called a couple times in the init lifecycle but because this is idempotent/can be applied multiple times to the same config I think it should be fine